optuna/study.py

```diff
@@ -848,6 +848,49 @@ def objective(trial):
 
         self._storage.create_new_trial(self._study_id, template_trial=trial)
 
+    @experimental("2.5.0")
+    def add_trials(self, trials: Sequence[FrozenTrial]) -> "Study":
+        """Add trials to study.
+
+        The trials are validated before being added.
+
+        Example:
+
+            .. testcode::
+
+                import optuna
+                from optuna.distributions import UniformDistribution
+
+
+                def objective(trial):
+                    x = trial.suggest_uniform("x", 0, 10)
+                    return x ** 2
+
+
+                study = optuna.create_study()
+                study.optimize(objective, n_trials=3)
+                assert len(study.trials) == 3
+
+                other_study = optuna.create_study().add_trials(study.trials)
+                assert len(other_study.trials) == len(study.trials)
+
+                other_study.optimize(objective, n_trials=2)
+                assert len(other_study.trials) == len(study.trials) + 2
+
+        .. seealso::
+
+            See :func:`~optuna.study.add_trial` for addition of each trial.
+
+        Args:
+            trials: Trials to add.
+
```

Comment:
How about adding `Returns:` and `Raises:` sections here?

Reply:
Thank you! I updated the PR following your comment.